From 0aaa35c6d3b07a273af6a1be2d26add4a9d537cd Mon Sep 17 00:00:00 2001
From: Noel Power <noel.power@novell.com>
Date: Thu, 1 Mar 2012 12:10:56 +0000
Subject: [PATCH 2/2] fix crash using instances dialog of dataform navigator
 fdo#44816

also silence a valgrind  'Conditional jump or move depends on uninitialised value(s)' where initial state of a checkbox was undefined
(cherry picked from commit 7b860d4970604f08ebd2e818bfd63891dd940804)

Signed-off-by: Michael Stahl <mstahl@redhat.com>
---
 svx/source/form/datanavi.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/svx/source/form/datanavi.cxx b/svx/source/form/datanavi.cxx
index e2e0b17..6718cd1 100644
--- a/svx/source/form/datanavi.cxx
+++ b/svx/source/form/datanavi.cxx
@@ -306,7 +306,8 @@ namespace svxform
         m_pNaviWin		( _pNaviWin ),
         m_bHasModel		( false ),
         m_eGroup		( _eGroup ),
-        m_TbxImageList	( SVX_RES( IL_TBX_BMPS ) )
+        m_TbxImageList  ( SVX_RES( IL_TBX_BMPS ) ),
+        m_bLinkOnce     ( false )
 
     {
         FreeResource();
@@ -1883,7 +1884,6 @@ namespace svxform
                             m_aTabCtrl.SetPageText( nId, sNewName );
                             bIsDocModified = true;
                         }
-                        delete(pPage);
                     }
                     break;
                 }
-- 
1.7.7.6

